Update GUI

jehousoh opened the pull request:
- Add 3 columns below name (Personal Particulars, Room Details, Room Status)
- Add icons to each label for less wordy Ui
- Add tooltips for each label for user to know what the label represents if they are unable to tell from icon
- Change colour of column headers to `#4ACFAC`
- Change application icon of the GuestBook
- Change isRoomClean to a green/red button for more intuitive yes/no

Closes #91

jehousoh committed:
Add icons to each field

Ui was very wordy due to all the information

Let's add icons to replace some of the words and make it intuitive

jehousoh committed:
Add room status column

jehousoh committed:
Fix text going out of screen

Let's implement text wrapping

jehousoh committed:
Update icon

jehousoh committed:
Change icon size

jehousoh committed:
Change icon size

jehousoh committed:
Merge branch 'master' into update-GUI

Add:
Room Cleaned to UI

Update:
Shift Number Of Guests to "Room Details"

jehousoh committed:
Merge branch 'master' into update-GUI

jehousoh committed:
Add bill to UI

jehousoh committed:
Fix bug where loading json file gives notClean image to all guests

jehousoh committed:
Merge branch 'master' into update-GUI

jehousoh committed:
Resize icon

jehousoh committed:
Update website and UG

jehousoh committed:
Merge branch 'master' into update-GUI
Add request to Ui

jehousoh committed:
Merge branch 'master' into update-GUI

jehousoh committed:
Add tooltips and removed label text

jehousoh committed:
Add gap between each label

jehousoh committed:
Update Ui.png

jehousoh committed:
Decrease gap between labels

jehousoh committed:
Add spacing between icon and text

bryanhce commented on docs/index.md:
```diff
@@ -1,6 +1,6 @@
 ---
 layout: page
-title: GuestBook Level-3
+title: GuestBook
```
Should the below be "desktop application for managing your guests' details" instead of "contact details"?

jehousoh replied:
Agreed and changed.

kangzongxian commented on src/main/java/seedu/guest/ui/GuestCard.java:
```diff
@@ -54,14 +59,112 @@ public GuestCard(Guest guest, int displayedIndex) {
         id.setText(displayedIndex + ". ");
         name.setText(guest.getName().fullName);
+        name.setWrapText(true);
+        Tooltip nameTT = new Tooltip("Name");
```
Should we abstract out the string values (e.g Name, Phone Number, etc...) as variables to make the code easier to maintain (e.g if the user wants to change the Field, they can just change it via the variable declared)

jehousoh replied:
Agreed. Created variables for each field.

kangzongxian commented on src/main/java/seedu/guest/ui/GuestCard.java:
```diff
@@ -54,14 +59,112 @@ public GuestCard(Guest guest, int displayedIndex) {
+        phone.setGraphic(setPhoneImage);
+        phone.setGraphicTextGap(7);
+        phone.setWrapText(true);
+        Tooltip phoneTT = new Tooltip("Phone Number");
```
Should we spell out TT fully as ToolTip for better clarity?

jehousoh replied:
Agreed and changed.

kangzongxian commented on src/main/java/seedu/guest/ui/GuestCard.java:
```diff
@@ -54,14 +59,112 @@ public GuestCard(Guest guest, int displayedIndex) {
+        Tooltip emailTT = new Tooltip("Email");
+        email.setTooltip(emailTT);
+
         room.setText(guest.getRoom().value);
```
Should we abstract the functionality of the various sections into methods so that it is cleaner and more maintainable?

jehousoh replied:
Agreed. Abstracted into a method.

kangzongxian reviewed (changes requested):
Looks good to me with some plausible minor changes!

bryanhce commented on src/main/java/seedu/guest/ui/GuestCard.java:
```diff
@@ -54,14 +59,112 @@ public GuestCard(Guest guest, int displayedIndex) {
+            setIsRoomCleanStatusImage.setFitHeight(18);
+            isRoomCleanStatus.setGraphic(setIsRoomCleanStatusImage);
+        } else {
+            Image image = new Image(getClass().getResourceAsStream("/images/roomNotClean.png"));
```
Should the name of the image be roomUnclean instead of NotClean to standardise the naming convention?

jehousoh replied:
Agreed and changed.

bryanhce commented on src/main/java/seedu/guest/ui/GuestCard.java:
```diff
@@ -54,14 +59,112 @@ public GuestCard(Guest guest, int displayedIndex) {
         phone.setText(guest.getPhone().value);
+        Image phoneImage = new Image(getClass().getResourceAsStream("/images/phone.png"));
+        ImageView setPhoneImage = new ImageView(phoneImage);
+        setPhoneImage.setFitWidth(16);
```
Should you use a variable to store the value of 16 in setFitWidth instead of hard coding the number? As the value 16 is repeated multiple times, it might be easier to abstract it out so that it will be easier to edit in the future.

jehousoh replied:
Agreed and changed.

bryanhce commented on src/main/java/seedu/guest/ui/GuestCard.java:
```diff
@@ -54,14 +59,112 @@ public GuestCard(Guest guest, int displayedIndex) {
+        setPhoneImage.setFitWidth(16);
+        setPhoneImage.setFitHeight(16);
+        phone.setGraphic(setPhoneImage);
+        phone.setGraphicTextGap(7);
```
Likewise with the value 7, should you abstract it into a variable too?

jehousoh replied:
Agreed and changed.

bryanhce commented on src/main/java/seedu/guest/ui/GuestCard.java:
```diff
@@ -54,14 +59,112 @@ public GuestCard(Guest guest, int displayedIndex) {
         this.guest = guest;
         id.setText(displayedIndex + ". ");
         name.setText(guest.getName().fullName);
+        name.setWrapText(true);
```
Just a suggestion, I see a couple of repeated fields, would it be possible to have there was a method created to build these ui components?

jehousoh replied:
Agreed. Abstracted into a method.

bryanhce reviewed (changes requested):
Nice work! Just some nits to fix.

jehousoh committed:
Improve abstracted

Abstracted each Label's text and graphic setting into a method
Change roomNotClean to roomUnclean for consistency

jehousoh committed:
Change contact to guests'

jehousoh committed:
Fix typo

bryanhce reviewed (approved):
LGTM

jehousoh committed:
Merge branch 'master' into update-GUI

kangzongxian reviewed (approved):
Looks good to me!

jehousoh committed:
Merge branch 'master' into update-GUI
